### ISSUE TITLE
[aio] SwUpdatesService never instantiates

### ISSUE THREAD

**Issue by tongkl1** [angular/angular#43695]
### Description

After a simple check I found SwUpdatesService is only "provided" in SwupdatesModule and never instantiates. I wonder whether this is intentional or is a bug.

### What is the affected URL?

Of course this applies to the whole site

### Please provide the steps to reproduce the issue

Simply add console.log() in SwUpdatesService::constructor() and it won't be printed when accessing the page.

After adding SwUpdatesService as dependency in an arbitrary class, the log info prints as expected.

### Please provide the expected behavior vs the actual behavior you encountered

SwUpdatesService instantiates and performs checks for update.

### Please provide the exception or error you saw

```true
SwUpdatesService doesn't instantiate.
```

### Is this a browser-specific issue? If so, please specify the device, browser, and version.

```true
No.
```

**Comment by gkalpak** [comment]
I assume you refer to the [SwUpdate][1] service. If that's the case, then the behavior you see is expected. This is how [Dependency Injection][2] works (i.e. the service is only instantiated if it is needed by another part of the app).

> SwUpdatesService instantiates and performs checks for update.

That is not exactly true :grin:
The `SwUpdate` service can be used to get notified about (and activate) updates and also to request a check for updates. But it is not necessary for the ServiceWorker to work (or check for updates).

The ServiceWorker checks for updates on its own (when it is initialized and on every navigation request). `SwUpdate` is there mainly to get notified about available updates (and to trigger update checks _in addition_ to those made by the ServiceWorker on its own).

In other words:
- Yes, the `SwUpdate` service will only be instantiated if your app uses it.
- No, the `SwUpdate` service is _not_ necessary for the ServiceWorker to work correctly and check for updates behind the scenes.

##
Closing, since everything works as intended, but feel free to continue the discussion below.


[1]: https://angular.io/api/service-worker/SwUpdate
[2]: https://angular.io/guide/architecture-services#dependency-injection-di

**Comment by gkalpak** [comment]
Oh, nevermind. I just realized you are talking about the [SwUpdatesService][1] in angular.io 😇
Let me look into that...

[1]: https://github.com/angular/angular/blob/c721135e370b34c840756bcfb22c8119b4c8c452/aio/src/app/sw-updates/sw-updates.service.ts

**Comment by tongkl1** [comment]
@gkalpak Actually I'm talking about not SwUpdate but [SwUpdatesService](https://github.com/angular/angular/blob/master/aio/src/app/sw-updates/sw-updates.service.ts) in aio.

**Comment by gkalpak** [comment]
So, you are totally right, @tongkl1 :grin:
We are not actually using the `SwUpdateService` 🤦‍♂️

Fortunately, we still get updates (because the ServiceWorker checks on its own), but we miss on applying them without a reload or handling `unrecoverable` events. Leaving this open to track the issue and decide what we want to do with it (remove `SwUpdatesService` or start using it).
